src/test/java/io/cdap/plugin/batch/source/FTPBatchSourceTest.java

```
@@ -162,4 +182,121 @@ public void testSFTPPathWithSystemPropertiesAndSpecialCharactersInAuth() {
   }
+
+  @Test
+  public void testInvalidServerFTPPathConnection() {
+    FailureCollector collector = new MockFailureCollector();
+    FTPBatchSource.FTPBatchSourceConfig config = new FTPBatchSource.FTPBatchSourceConfig();
+    FakeFtpServer server = ftpSetup();
+    config.configuration("ftp://user:password@invalid_server:21", "");
+    config.validate(collector);
+    server.stop();
+    Assert.assertTrue(collector.getValidationFailures().size() > 0);
+  }
+
+  @Test
+  public void testInvalidUsernamePasswordFTPPathConnection() {
+    FailureCollector collector = new MockFailureCollector();
+    FTPBatchSource.FTPBatchSourceConfig config = new FTPBatchSource.FTPBatchSourceConfig();
+    FakeFtpServer server = ftpSetup();
+    config.configuration("ftp://wronguser:wrongpassword@localhost:21", "");
+    config.validate(collector);
+    server.stop();
+    Assert.assertTrue(collector.getValidationFailures().size() > 0);
+  }
+
+  @Test
+  public void testValidFTPPathConnection() {
+    FailureCollector collector = new MockFailureCollector();
+    FTPBatchSource.FTPBatchSourceConfig config = new FTPBatchSource.FTPBatchSourceConfig();
+    FakeFtpServer server = ftpSetup();
+    config.configuration("ftp://user:password@localhost:21", "");
+    config.validate(collector);
+    server.stop();
+    Assert.assertEquals(0, collector.getValidationFailures().size());
+  }
+
+  @Test
+  public void testInvalidServerSFTPPathConnection() {
+    FailureCollector collector = new MockFailureCollector();
+    FTPBatchSource.FTPBatchSourceConfig config = new FTPBatchSource.FTPBatchSourceConfig();
+    try {
+      SshServer server = sftpSetup();
+      config.configuration("sftp://user:password@invalid_server:22", "");
+      config.validate(collector);
+      server.stop();
+      Assert.assertTrue(collector.getValidationFailures().size() > 0);
+    } catch (Exception e) {
+      // do nothing
```

Remove empty catch if we don't expect to get an exception